### DISCUSSION TITLE
Trying to use Wemos V4 board

### DISCUSSION THREAD

**Discussion by Alexd911** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#593] [Q&A]
Hey,

Firstly have to say great projet:)
So to the question and BTW I am totally new at the coding part.. Hardware side I'm OK with.

I did try and search for the V4 board, but didnt find anything specific to editing the Code or HW.

So I ended up buying the WEMOS V4 of the board not realy knowing what I was doing and after finally working out that the Pins were different and using https://escapequotes.net/esp8266-wemos-d1-mini-pins-and-diagram/ as a guide I think I managed to convert the pins:

D0 = 16
D5 = 14
D6 = 12
D7 = 13

D2 = 4
D3 = 0
D4 = 2

After making the board nothing worked, so after some thiniing I thought it could be the HW setup in the app, but the app did not only used Numbers and not prefixed with a "D" so I did not try entering just the number eg instead of Audio = 0 to enter 16.

I looked at the code and did find reference to the D0, D5 etc in BWC.CPP with Pin(0) = D1 etc but was not sure if I should mess with the code.

So any suggestions how to get the V4 board to work, or do I scrap it and get the versions shown in the guide?

Thanks in advance and as sai I am a total newbie, so may need to simple guide.

Cheers
Alex

**Comment by Alexd911** [comment]
I forgot to say that as the V4 uses 3.3V I just used the USB port to power it and by passed the 5V going to the board and just connected the + - from one connector to the other and spliced in a USB C connector to power the 5V.

Alex

**Comment by visualapproach** [comment]
I think you're over complicating it. All 8266 are 3.3V but most breakout boards have a 5V pin connected to a 3V3 regulator so I suggest you use it.
If you choose to not use the NodeMCU you have to edit the board parameter in platformio.ini. The correlation between Dx -> GPIOx is  then done automatically so you don't have to worry about GPIOs in this project. 
User interface update functionality will download firmware for NodeMCU so it won't work with boards with another pin mapping.
That said, D1 mini has the same Dx-GPIO mapping as NodeMCU as far as I know so the above is not necessary. I'm not familiar with the "v4" though.

Also if you are using a bare mini (or any other board) without a level converter it may or may not work.

**Reply by Alexd911** [reply]
Hey,

Big thanks for your help, but unfortunalty I can't see where in the platformio.ini I need to edit.

Here are a couple of pictures, so 1st one is from the website showing 5V and the next ones are from my board with 3.3v

![image](https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/assets/142144200/b246f24f-56a9-4c0d-8851-d5fcdb8581b4)


As you can see mine has 3.3V vs 5v from the one in the project.

What I am unsure is the Pin numbering from the Original projct pictures and set up, to the one I have.

Do I need to edit platformio.ini as you mentioned and if so what do I edit as I am using the latest build where you do not edit the file for the 6 pin, rather you do that in the user interface.

![IMG_6406](https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/assets/142144200/7329f128-a2c6-4233-b884-6bc6074762bd)

![IMG_6405](https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/assets/142144200/23eb7391-1d42-4fd9-bb68-92eed40f2894)

As mentioned I have very limited knowldege with VS.

Once again I appreciate your help and sorry for the stupid questions.

Regards

Alex

**Reply by visualapproach** [reply]
Reading up on v4 reveals that it's pin compatible with earlier versions. They write the gpio numbers on the silkscreen instead of D#. VBUS is 5V input. So treat it as a v3.

**Reply by Alexd911** [reply]
Hey

Thanks for this information. 
Cheers for the hint on the VBUS and I missed that when reading about the v4.
I will rewire that so I can drop the USB-C, but meanwhile I tries it again but no joy:(
When re wiring it for the 5V I will check the solder joints again in case any are bad.

Below are the settings I used and I think they are correct.
I am using the same pump as in the guide.

<img width="434" alt="Screenshot 2023-08-14 at 21 30 37" src="https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/assets/142144200/e53c97c0-be67-4f40-895b-0a3eb3ea6402">

When powering back up I do not get anything on the dispaly and not even a beep.

Once again I really appreciate your help and guidance its very appreciated.

Regards

Alex

**Reply by visualapproach** [reply]
You can't use gpio numbers. There is no D16. You have to use the Dx format which you can see on d1 mini v3

**Comment by metaljay** [comment]
Just to add, this seems exactly like: https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/issues/533

conclusion, the LLC is required. Trust me you’ll spend ages trying to get it to work and in the end just order one of visual approaches boards.

falling that I’m selling some in the discussions tab: 
https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/discussions/580

**Reply by Alexd911** [reply]
Hey,

Sorry for the dumn question, but what is a LLC?

Is the board alone not enough as shown in the guide as it seems that you just plug it in with no extra hardware.

Sorry for the question..

Alex

**Reply by metaljay** [reply]
low level converter.
I like you followed that guide and bought a D1 mini, if you scroll down in the Link I gave you above you'll see all my photos too, in short the guy who wrote that guide has a different pump to us and didn't need it, modern pumps require the LLC.

**Reply by visualapproach** [reply]
> as it seems that you just plug it in with no extra hardware.

You don't see ME plugging it in without LLC in my manual! There is a discussion about barebone mini D1 and there is a third party guide out there, made by someone else (a very nice guide btw) but it's not according to my manual. You are more than welcome to connect to anything you want, but I agree with metaljay - if shortcut doesn't work, take the intended way. I may sound rude here. Thats not my intention. I just want to be clear. I always try to help as much as I can by answering questions but I can't put too much time explaining how to do things "out of spec".

**Reply by Alexd911** [reply]
Hey all,


Thanks to everyone and seems I simplified the project in my head too much and thought I could use the Wemos D1 mini V4 on its own.
WHen looking at the pictures it just seemed simple to just connect it in place, but as you mentioned I must be using a different pump.
My pump is the S100101, so in that case I would need the LLC I suppose. 

Can you propose a LLC model so next time I am ordering from Aliexpress I can buy one.

Cheers.

Alex

**Reply by visualapproach** [reply]
Sure thing: https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/blob/master/bwc-manual.pdf

**Reply by Alexd911** [reply]
Hey,

I dont mean to upset anyone, but I am still somewhat confussed even after all the great help I have had.

In the guide from here https://antanaitis.lt/2022/09/07/wi-fi-on-your-lay-z-spa-make-it-easy-to-remote-control/  which is a very good and clear guide BTW, I do have the exact same setup using the same pump (S100101), but using the wemos D1_Mini V4 and only refernce that I dont know is the reference to the "board with 5V pinout". I honestly cant see any reference to the LLC or the PCB that has been shown in some other posts.
I am obviously missing something in the guide and just too stupid to spot it..
Again I am sorry as I dont want to take anyones time away to answer something that should be so apparent.

Big thanks.
Alex

**Reply by metaljay** [reply]
did you read the link I gave you? you'll see various posts including S100101 and S100102 whom all have the same problem and require a LLC
